backend/src/models/email_verification_request.rs

```diff
@@ -0,0 +1,130 @@
+use crate::entities::email_verification_request;
+use crate::models::user::UserInformation;
+use crate::utils::error::Error;
+use sea_orm::{DatabaseConnection, EntityTrait, Set};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use uuid::Uuid;
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "snake_case", tag = "name", content = "data")]
+pub enum EmailVerificationAction {
+    ConfirmEmailAddress { user_information: UserInformation },
+    ResetPassword { email: String },
+}
+
+impl EmailVerificationAction {
+    pub fn get(&self) -> (String, Option<Value>) {
+        let value = serde_json::to_value(self).unwrap();
+        let name = value.get("name").unwrap().as_str().unwrap().to_string();
+        let data = value.get("data").cloned();
+
+        (name, data)
+    }
+}
+
+impl email_verification_request::Model {
+    pub async fn create(
+        database: &DatabaseConnection,
+        action: EmailVerificationAction,
+        email: String,
+        expiration_duration: Option<Duration>,
+    ) -> Result<Uuid, Error> {
+        let (action_type, action_data) = action.get();
+        let confirmation_code = Uuid::new_v4();
+
+        email_verification_request::Entity::insert(email_verification_request::ActiveModel {
+            id: Set(confirmation_code),
+            action_type: Set(action_type),
+            additional_data: Set(action_data),
+            email: Set(email),
+            expiration_time: Set(
+                expiration_duration.map(|duration| Utc::now().naive_utc().add(duration))
+            ),
+            created_at: Set(chrono::Utc::now().naive_utc()),
+        })
+        .exec(database)
+        .await?;
+
+        Ok(confirmation_code)
+    }
+
+    pub async fn get(database: &DatabaseConnection, id: Uuid) -> Result<Self, Error> {
```

Comment:
```suggestion
    pub async fn find_by_id(database: &DatabaseConnection, id: Uuid) -> Result<Self, Error> {
```
or something, because find by id is entity method, anyway it should match names in other struct, at lest this is what you said to me :3

Reply:
This method does a lot more than only finding by id, but I agree that `get` isn't the best name